types/prosemirror-view/index.d.ts

```diff
@@ -683,14 +687,22 @@ export interface NodeView<S extends Schema = any> {
   /**
    * Called when a DOM
    * [mutation](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)
-   * happens within the view. Return false if the editor should
+   * or a selection change happens within the view. When the change is
+   * a selection change, the record will have a `type` property of
+   * `"selection"` (which doesn't occur for native mutation records).
+   * Return false if the editor should re-read the selection or
    * re-parse the range around the mutation, true if it can safely be
    * ignored.
    */
-  ignoreMutation?: ((p: MutationRecord) => boolean) | null;
+  ignoreMutation?: ((p: MutationRecord & { readonly type: ProsemirrorMutationRecordType }) => boolean) | null;
   /**
    * Called when the node view is removed from the editor or the whole
    * editor is destroyed.
    */
   destroy?: (() => void) | null;
 }
+/**
+ * When the change is a selection change, the record will have a `type` property
+ * of `"selection"` (which doesn't occur for native mutation records).
+ */
+export type ProsemirrorMutationRecordType = 'attributes' | 'characterData' | 'childList' | 'selection';
```

Comment:
FYI: The following is also possible:

```typescript
export type ProsemirrorMutationRecordType = MutationRecord['type'] | 'selection';
```

This makes a little bit clearer what's added.

Reply:
True 👍